src/models/room.ts

```diff
@@ -913,9 +913,10 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
         const backwardPaginationToken = liveTimelineBefore.getPaginationToken(EventTimeline.BACKWARDS);
         const eventsBefore = liveTimelineBefore.getEvents();
         const mostRecentEventInTimeline = eventsBefore[eventsBefore.length - 1];
+        const mostRecentEventIdInTimeline = mostRecentEventInTimeline?.getId();
```

Comment:
I am surprised that TypeScript didn't catch this before. I would have assumed that TypeScript would infer `undefined` as a possibility from an array lookup but I guess not.

Related https://stackoverflow.com/questions/59479171/typescript-does-not-infer-undefined-as-a-return-type-from-array-prototype-find which mentions that `strictNullChecks` needs to be enabled for it to infer `undefined` there.

Is `strictNullChecks` separate from `strict`? We added `strict` recently in https://github.com/matrix-org/matrix-js-sdk/pull/2835

Related to https://github.com/vector-im/element-web/issues/21967 for adding `strictNullChecks`